### PR TITLE
Send an alert to PagerDuty when accessor token is about to expire

### PR DESCRIPTION
**Pull request by eherot**

### CODE REVIEW

File: resources/replace_vault_token.rb [hunk 1]
```diff
@@ -54,20 +57,37 @@ def self.valid_url(url)
...
+    if (ttl = v.auth_token.lookup_self.data[:ttl]) < (3_600 * 24 * 3) && ttl != 0
+      Chef::Log.warn "The accessor token for #{token_name} expires in #{ttl} seconds"
+      alert_token_expiring ttl if pagerduty_key
+    end
+    v
   end
 end
 
+def alert_token_expiring(ttl)
+  msg = "The accessor token used for replacing #{token_name} expires in less than 3 days"
```
Comment (jeffbyrnes) [inline resources/replace_vault_token.rb]: I think it’d be stronger if this and the Chef log message more clearly called out the resolution: regenerate a token with the same privileges.

**Comment by jeffbyrnes** [comment]
@eherot awesome, thanks for tackling this ASAP. @andrhamm I think this will solve the pain we discussed yesterday.

**Comment by jeffbyrnes** [comment]
@eherot and to be clear, other than my comment about the messaging, this looks good.

File: resources/replace_vault_token.rb (hunk 1 above)
Comment (eherot) [inline resources/replace_vault_token.rb]: Keep in mind this is an error in a generic resource. The correct policy name is not necessarily known (but you can look it up for the token in question if it hasn't expired yet).

Comment (jeffbyrnes) [inline resources/replace_vault_token.rb]: Very true. Doesn’t need to specify the exact policy, but I think should still direct the operator that that is the action to take.

File: resources/replace_vault_token.rb [hunk 2]
```diff
@@ -54,20 +57,38 @@ def self.valid_url(url)
...
+      Chef::Log.warn "The accessor token for #{token_name} expires in #{ttl} seconds"
+      alert_token_expiring ttl if pagerduty_key
+    end
+    v
   end
 end
 
+def alert_token_expiring(ttl)
+  msg = "The accessor token used for replacing #{token_name} expires in less than 3 days. " \
+    'Please replace this token with a new one using the token/create API.'
```
Comment (jeffbyrnes) [inline resources/replace_vault_token.rb]: Great! Thanks much.

**Pull request merged**